indico/modules/events/editing/client/js/editing/timeline/StateIndicator.jsx

```diff
@@ -18,33 +18,41 @@ const colors = {
   needs_submitter_changes: 'orange',
 };
 
-export default function StateIndicator({text, circular, tooltip, state}) {
-  const trigger = (
-    <Label size="tiny" color={colors[state]} circular={circular}>
-      {text}
-    </Label>
-  );
+const labelText = {
+  replaced: 'Replaced',
+  needs_submitter_confirmation: 'Needs submitter confirmation',
+  rejected: 'Rejected',
+  accepted: 'Accepted',
+  assigned: 'Assigned',
+  needs_submitter_changes: 'Needs submitter changes',
+};
+
+export default function StateIndicator({label, circular, tooltip, state}) {
+  const trigger = <Label size="tiny" color={colors[state]} circular={circular} />;
 
   return (
-    <Popup
-      position="bottom center"
-      trigger={trigger}
-      content={tooltip}
-      on="hover"
-      disabled={!tooltip}
-    />
+    <>
+      <Popup
+        position="bottom center"
+        trigger={trigger}
+        content={tooltip}
+        on="hover"
+        disabled={!tooltip}
+      />
+      {label && <div style={{marginLeft: '5px', color: colors[state]}}>{labelText[state]}</div>}
```

Comment:
For longer text it looks like this 
![image](https://user-images.githubusercontent.com/17234430/81063322-df148c00-8ed7-11ea-8ed0-41cdfe8ead38.png)

One option would be to use `white-space: nowarp; text-overflow: ellipsis; overflow: hidden;` and it would look like this:
![image](https://user-images.githubusercontent.com/17234430/81063573-5ea25b00-8ed8-11ea-9893-f8c889f407c1.png)

Maybe adding a tooltip would help in this case also. 

Other option would be to make the column wider maybe?

Reply:
I will use this: `white-space: nowarp; text-overflow: ellipsis; overflow: hidden;`

Nice catch :sweat_smile:

Reply:
But then you need the tooltip for sure.

Reply:
I'm wondering if it would be possible to make one of the columns grow, perhaps using something such as `flexGrow={1}`. In the initial example, the title has an excessive width of `600px`, when we could reduce and shrink it (`flexShrink`) so the content fits without wrapping. Would this work?

Reply:
When we will have actual data, the title will be long enough. I don't think then we need this.